source/codegen/templates/service.cpp.mako

```diff
@@ -37,6 +37,20 @@ namespace ${config["namespace_component"]}_grpc {
   {
   }
 
+% if common_helpers.has_bytes_type_param(functions):
+  template <typename T1, typename T2>
+  void ${service_class_prefix}Service::Copy(const std::vector<T1>& input, T2 output)
```

Comment:
I have let it to be `T2 output` only as while calling the Copy function I am passing typecasted pointer, `Copy(an_array, (ViUInt8*)response->mutable_an_array()->data());` 
So now in place of T2 (ViUInt8*) will be taken and I can iterate over the output. The tests are working fine with this.

Reply:
I don't see your changes here, but if you use `T2` as type in the template, you can't really do `output[i]` since from template perspective `T2` need not be pointer. Users can pass `Int32` as T2 and it would be valid syntactically. By marking it as `T2*`, template explicitly says it expects pointer to data.

Reply:
Yes, now it seems logical to me why T2* is to be used because we want this Copy function to expect pointer type as 2nd parameter so that we can use it to iterate over output. Thanks, I have made the change.

Reply:
Resolving based on our recent discussion.